pathfinding_sdf/pathfinding_sdf.c

```diff
@@ -0,0 +1,670 @@
+*   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+*   copies of the Software, and to permit persons to whom the Software is
+*   furnished to do so, subject to the following conditions:
+*
+*   The above copyright notice and this permission notice shall be included in all
+*   copies or substantial portions of the Software.
+*
+*   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+*   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+*   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+*   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+*   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+*   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+*   SOFTWARE.
+*
+**********************************************************************************************
+*
+*   This example demonstrates how to use signed distance fields (SDF) for improving
+*   pathfinding.
+*
+*   1) Unit size: A path may have a requirement for a minimum width to allow passage.
+*      By using the SDF values, we can block paths that are too narrow for the unit.
+*
+*   2) Path preferences: A unit may prefer to stay close to walls or avoid them.
+*      The example demonstrates how to influence the pathfinding by using the SDF values.
+*
+*   3) Varying step distances: Using SDF values to adjust step distances during path 
+*      finding, resulting in curved paths.
+*
+*  TODO: optimize the path after it was found: nodes could be removed to smooth the path
+*        this could again be done using the SDF values to determine if a node can be removed
+*        without clipping through walls
+*
+********************************************************************************************/
+
+#include "raylib.h"
+#include "raymath.h"
+
+#include <stddef.h> // Required for: NULL
+
+typedef struct PathfindingNode
+{
+    int x, y;
+    int fromX;
+    int fromY;
+    int score;
+} PathfindingNode;
+
+typedef struct NeighborOffset
+{
+    int x, y;
+    int distance;
+} NeighborOffset;
+
+// a simple cat face that can be drawn as a triangle fan
+Vector2 catFace[] = {
+    {0.0f, 1.0f},
+    {0.7f, 1.0f},
+    {1.0f, 0.7f},
+    {1.0f, -1.0f},
+    {0.5f, -0.6f},
+    {-0.5f, -0.6f},
+    {-1.0f, -1.0f},
+    {-1.0f, 0.7f},
+    {-0.7f, 1.0f},
+};
+
+// a simple rat face that can be drawn as a triangle fan
+Vector2 ratFace[] = {
+    {0.0f, 1.0f},
+    {0.3f, 0.9f},
+    {0.8f, -0.2f},
+    {1.0f, -0.8f},
+    {0.8f, -1.0f},
+    {0.5f, -1.0f},
+    {0.3f, -0.7f},
+    {0.0f, -0.8f},
+    {-0.3f, -0.7f},
+    {-0.5f, -1.0f},
+    {-0.8f, -1.0f},
+    {-1.0f, -0.8f},
+    {-0.8f, -0.2f},
+    {-0.3f, 0.9f},
+};
+
+const int gridWidth = 80;
+const int gridHeight = 45;
+const int cellSize = 10;
+char* blockedCells;
+char* sdfCells;

```

Comment:
Please init your pointers to null

Reply:
Fixed, I hope to haven't overlooked any. It's my habit to not initialize global variables since I know they default to 0 / NULL, but you are correct that this is cleaner.